src/timely-util/src/builder_async.rs

```diff
@@ -212,6 +219,110 @@ impl<'handle, T: Timestamp, D: Container, P: Pull<BundleCore<T, D>>> Future
     }
 }
 
+// TODO: delete and use CapabilityTrait instead once TimelyDataflow/timely-dataflow#512 gets merged
+pub trait CapabilityTrait<T: Timestamp> {
+    fn session<'a, D, P>(
```

Comment:
You might want to add `#[inline]` to all implementations of `session`. We're not doing this much in Mz, but it seems we might want to start, especially when functions are called across crate boundaries.